simpleclient_hdrhistogram/src/main/java/io/prometheus/client/HdrSummary.java

```diff
@@ -0,0 +1,399 @@
+ *         .name("requests_latency_seconds").help("Request latency in seconds.").register();
+ *
+ *     void processRequest(Request req) {
+ *       HdrSummary.Timer requestTimer = requestLatency.startTimer();
+ *       try {
+ *         // Your code here.
+ *       } finally {
+ *         receivedBytes.observe(req.size());
+ *         requestTimer.observeDuration();
+ *       }
+ *     }
+ *
+ *     // Or if using Java 8 and lambdas.
+ *     void processRequestLambda(Request req) {
+ *       receivedBytes.observe(req.size());
+ *       requestLatency.time(() -> {
+ *         // Your code here.
+ *       });
+ *     }
+ *   }
+ * }
+ * </pre>
+ * This would allow you to track request rate, average latency and average request size.
+ *
+ * <p>
+ * How to add custom quantiles:
+ * <pre>
+ * {@code
+ *   static final HdrSummary myMetric = HdrSummary.build()
+ *       .quantile(0.5)  // Add 50th percentile (= median)
+ *       .quantile(0.9)  // Add 90th percentile
+ *       .quantile(0.99) // Add 99th percentile
+ *       .name("requests_size_bytes")
+ *       .help("Request size in bytes.")
+ *       .register();
+ * }
+ * </pre>
+ *
+ * The quantiles are calculated over a sliding window of time. There are two options to configure this time window:
+ * <ul>
+ *   <li>maxAgeSeconds(long): Set the duration of the time window is, i.e. how long observations are kept before they are discarded.
+ *       Default is 10 minutes.
+ *   <li>ageBuckets(int): Set the number of buckets used to implement the sliding time window. If your time window is 10 minutes, and you have ageBuckets=5,
+ *       buckets will be switched every 2 minutes. The value is a trade-off between resources (memory and cpu for maintaining the bucket)
+ *       and how smooth the time window is moved. Default value is 5.
+ * </ul>
+ *
+ * See https://prometheus.io/docs/practices/histograms/ for more info on quantiles.
+ */
+public class HdrSummary extends SimpleCollector<HdrSummary.Child> implements Counter.Describable {
+
+  private final List<Double> quantiles; // Can be empty, but can never be null.
+  private final long highestToLowestValueRatio;
+  private final int numberOfSignificantValueDigits;
+  private final long maxAgeSeconds;
+  private final int ageBuckets;
+
+  private HdrSummary(Builder b) {
+    super(b);
+    this.quantiles = Collections.unmodifiableList(new ArrayList<Double>(b.quantiles));
+    this.highestToLowestValueRatio = b.highestToLowestValueRatio;
+    this.numberOfSignificantValueDigits = b.numberOfSignificantValueDigits;
+    this.maxAgeSeconds = b.maxAgeSeconds;
+    this.ageBuckets = b.ageBuckets;
+    initializeNoLabelsChild();
+  }
+
+  public static class Builder extends SimpleCollector.Builder<Builder, HdrSummary> {
+
+    private final List<Double> quantiles = new ArrayList<Double>();
+    private long highestToLowestValueRatio = 1000;
+    private int numberOfSignificantValueDigits = 2;
```

Comment:
I'll follow up regarding defaults in https://github.com/HdrHistogram/HdrHistogram/issues/156.

Reply:
I decreased the default initial dynamic range to `1000` and the default number of significant value digits to `2` based on previous suggestions.